src/ScaleIdentification.jl

```diff
@@ -0,0 +1,107 @@
+using MIDI
+
+
+"""
+    MIDI_to_notes
+
+Returns the note names of an array of MIDI pitch value, inside of single octave.
+
+Example :
+julia> Midi_to_notes([0,13,26])
+["C","C","C"]
+
+"""
+function MIDI_to_notes(MIDInotes::Array{})
+    name = ["C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B"]
+    MIDIrange = []
+    for i in range(-1,12)
+        for note in name
+            push!(MIDIrange,note)
+        end
+    end
+    notes = []
+    for n in MIDInotes
+        push!(notes,MIDIrange[n+1])
+    end
+    return notes
+end
+
+
+"""
+    most_frenquent_notes(notes)
+
+Takes an Array representing musical notes
+and returns the 7 most frequent values from this array.
+
+"""
+function most_frenquent_notes(notes::Array{})
+    occurency = sort(Dict(value => key for (key, value) in countmap(notes)), rev = true)
+    most_frequent = Any[n for n in values(occurency)]
+    return most_frequent[1:7]
+end
+
+"""
+    scale_identification(scales, notes::Array{})
+
+Takes an Array of MIDI pitches (ranging from 0 to 127) and returns the most probable scale it belongs to.
+If the scale is unknown or the piece contains several tonality, will return : 
+"Unregistered exotic scale or atonal/modulating musical piece"
+
+"""
+function scale_identification(scales, MIDInotes::Array{})
+    notes = MIDI_to_notes(MIDInotes)
+    for (k,v) in scales
+        tester = true
+        for n in most_frenquent_notes(notes)
+            if n ∉ v
+                tester = false
+            end
+        end
+        if tester == true
+            return k,v
+            break
+        end
+    end
+    if tester == false
+        print("Unregistered exotic scale or atonal/modulating musical piece")
+    end
+end
+
+scales = Dict()
```

Comment:
`scales` _must_ be defined with a **concrete type** (`Dict{String, Vector{String}`) and it must be defined as a `const`.